core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/util/DefaultIItemResultSet.java

```diff
@@ -134,7 +134,8 @@ public void run() {
 					try {
 						if (!exec.awaitTermination(60, TimeUnit.SECONDS)) {
 							exec.shutdownNow();
-							exec.awaitTermination(configuredTimeout, TimeUnit.MINUTES);
+							if (configuredTimeout != 0)
+								exec.awaitTermination(configuredTimeout, TimeUnit.MINUTES);
```

Comment:
Sorry looking at this now I think the proposition (1) in [comment](https://github.com/openjdk/jmc/pull/419#discussion_r988313757) is incorrect

* ~~a guard value like `-1` is better and in `DefaultIItemResultSet` just skip the termination.~~ 

If the value is `0` (or the `-1` which conveys to human this is an invalid value and a such disable the timeout) the code should just wait that all results have been delivered. Also maybe make this guard value a constant that can be reused in `ItemResultSetFactory`.

Instead the other proposition, to have a very large termination timeout, might be more appropriate in the present case, JDK 19's `ExecutorService.close()` for example use a timeout of _1 day_.